### ISSUE TITLE
Support for SCSS/SASS Syntax or Option to Ignore Specific Selectors/Properties

### ISSUE THREAD

**Issue by erwanjugand** [ryo-manba/stylelint-plugin-use-baseline#20]
Hello, and thank you for this great Stylelint plugin!

While testing the plugin, I encountered an issue with SCSS/SASS syntax. Specifically, it triggers the error:
`Selector "nesting" is not a widely available baseline feature`.

Would it be possible to support SCSS/SASS syntax?

Alternatively, could you provide an option to define a list of Selectors or Properties to ignore?

For example:
```json
"rules": {
    "plugin/use-baseline": [
      true,
      {
        "available": "newly",
        "ignoreSelectors": ["Nesting"],
        "ignoreProperties": ["text-wrap"]
      }
    ]
}
```

Thank you for your consideration!

**Comment by jeddy3** [comment]
I'm not the maintainer of this plugin, but I can chime in a suggestion.

> Selector "nesting" is not a widely available baseline feature.

This is flagging the [CSS nesting feature](https://drafts.csswg.org/css-nesting/#nest-selector).

Adding `ignoreSelectors` and `ignoreProperties` options to make the rule more permissive would align with how [we do it for Stylelint's built-in rules](https://stylelint.io/developer-guide/rules#user-defined-ignore).

People who are preprocessing their CSS (whether through a language extension like SCSS or a tool like [postcss-present-env](https://preset-env.cssdb.org/)) can then ignore specific checks.

> Would it be possible to support SCSS/SASS syntax?

The plugin _generally_ supports SCSS syntax because the stylesheet is broadly parsed with PostCSS (or one of its custom syntaxes). CSSTree, which only supports CSS, is only used for [smaller contexts](https://github.com/csstree/csstree/blob/master/docs/parsing.md#context) like selectors, at-rule preludes etc. It'll silently ignore encounters with any SCSS features that are invalid CSS, such as interpolation.

**Comment by ryo-manba** [comment]
Thanks for the help, @jeddy3!

Hi @erwanjugand,
Besides the nesting warning, do any other SCSS/SASS errors appear?
If so, please share a small code example.

**Comment by erwanjugand** [comment]
> Besides the nesting warning, do any other SCSS/SASS errors appear?

After testing on several SCSS projects, I haven’t encountered any other errors.
